builder/amazon/common/step_run_spot_instance.go

```diff
@@ -280,23 +278,22 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 		Type: aws.String("instant"),
 	}
 
-	var createOutput *ec2.CreateFleetOutput
-
+	req, createOutput := ec2conn.CreateFleetRequest(createFleetInput)
 	err = retry.Config{
 		Tries: 11,
 		ShouldRetry: func(err error) bool {
-			if strings.Contains(err.Error(), "Invalid IAM Instance Profile name") {
+			if IsAWSErr(err, "InvalidParameterValue", "iamInstanceProfile") {
 				// eventual consistency of the profile. PutRolePolicy &
 				// AddRoleToInstanceProfile are eventually consistent and once
 				// we can wait on those operations, this can be removed.
 				return true
 			}
-			return request.IsErrorRetryable(err)
+			return false
 		},
 		RetryDelay: (&retry.Backoff{InitialBackoff: 500 * time.Millisecond, MaxBackoff: 30 * time.Second, Multiplier: 2}).Linear,
 	}.Run(ctx, func(ctx context.Context) error {
-		createOutput, err = ec2conn.CreateFleet(createFleetInput)
-
+		// Actually send the spot connection request.
+		err := req.Send()
```

Comment:
Does using a request object changes something here ? The way I understood this that it changes nothing:

https://github.com/hashicorp/packer/blob/0fa60c68fbfcf5ef29485489a41f5afa0565bdee/vendor/github.com/aws/aws-sdk-go/service/ec2/api.go#L4071-L4072

```
// This method is useful when you want to inject custom logic or configuration
// into the SDK's request lifecycle. Such as custom headers, or retry logic.
```

Reply:
I don't exactly remember if this was the reason but I had to change do keep the instance list in createOutput, the other way I think was overriding the variable all the time and loosing the successfully created instances. But I'll have to check that again.

Reply:
Passing to confirm the reason I said above

Reply:
Passing to confirm what I said above.